pymatgen/io/vasp/inputs.py

```diff
@@ -419,16 +420,36 @@ def from_str(cls, data, default_names=None, read_velocities=True):
                 for i, nat in enumerate(n_atoms):
                     sym = Element.from_Z(i + 1).symbol
                     atomic_symbols.extend([sym] * nat)
-                warnings.warn(f"Elements in POSCAR cannot be determined. Defaulting to false names {atomic_symbols}.")
-        # read the atomic coordinates
+                warnings.warn(
+                    f"Elements in POSCAR cannot be determined. Defaulting to false names {atomic_symbols}.",
+                    BadPoscarWarning,
+                )
+
+        # Read the atomic coordinates
         coords = []
         selective_dynamics = [] if has_selective_dynamics else None
         for i in range(n_sites):
             tokens = lines[ipos + 1 + i].split()
             crd_scale = scale if cart else 1
             coords.append([float(j) * crd_scale for j in tokens[:3]])
             if has_selective_dynamics:
-                selective_dynamics.append([tok.upper()[0] == "T" for tok in tokens[3:6]])
+                # Warn when values contain suspicious entries
+                if any(value not in {"T", "F"} for value in tokens[3:6]):
+                    warnings.warn("Selective dynamics values must be either 'T' or 'F'.", BadPoscarWarning)
+
+                # Warn when elements contains Fluorine (F) (#3539)
+                if atomic_symbols[i] == "F" and len(tokens[3:]) >= 4 and "F" in tokens[3:7]:
```

Comment:
The warning is intended to be species, and is only issued when all three conditions are met:
1. The element for this line is Fluorine;
2. This line does have extra information (comment or such);
3. There is "F" within the suspicious range.

Reply:
Thanks @DanielYang59, sorry I missed this last week. This looks good!

Reply:
@esoteric-ephemera No rush at all. Thanks for reviewing. I was down for a whole week because of flu.